amethyst_core/src/transform/components/local_transform.rs

```diff
@@ -1,28 +1,38 @@
 //! Local transform component.
 
-use cgmath::{Deg, InnerSpace, Matrix3, Matrix4, Point3, Quaternion, Rotation, Rotation3, Vector3};
+use cgmath::{
+    Deg,
+    EuclideanSpace,
+    InnerSpace,
+    Matrix3,
+    Matrix4,
+    Point3,
+    Quaternion,
+    Rotation,
+    Rotation3,
+    Vector3
+};
 use orientation::Orientation;
 use specs::{Component, DenseVecStorage, FlaggedStorage};
 
 /// Local position, rotation, and scale (from parent if it exists).
 ///
 /// Used for rendering position and orientation.
-#[derive(Clone, Debug, PartialEq, PartialOrd)]
+#[derive(Clone, Debug, PartialEq)]
 pub struct LocalTransform {
     /// Quaternion [w (scalar), x, y, z]
-    pub rotation: [f32; 4],
+    pub rotation: Quaternion<f32>,
     /// Scale vector [x, y, z]
-    pub scale: [f32; 3],
+    pub scale: Vector3<f32>,
     /// Translation/position vector [x, y, z]
-    pub translation: [f32; 3],
+    pub translation: Point3<f32>,
```

Comment:
This should be Vector, translation is a movement quantity, but a positional quantity

Reply:
And it's a relative translation from the parent and not an absolute position

Reply:
I think the most compelling part is that it's a relative quantity, not absolute, in `LocalTransform`